rustls/src/quic.rs

```diff
@@ -171,7 +296,7 @@ pub(crate) fn read_hs(this: &mut ConnectionCommon, plaintext: &[u8]) -> Result<(
     Ok(())
 }
 
-pub(crate) fn write_hs(this: &mut ConnectionCommon, buf: &mut Vec<u8>) -> Option<Keys> {
+pub(crate) fn write_hs(this: &mut ConnectionCommon, buf: &mut Vec<u8>) -> Option<KeyMaterial> {
```

Comment:
Would it make sense to have two `write_hs` functions, the first which returns `Option<HandshakeKey>` and then second of which returns `Option<OneRttKey>`, where the QUIC implementation would call the first one until it returns Some, and then call the second one for the rest?

Reply:
That seems like a pretty awkward API. The current API seems pretty precisely typed -- I don't really understand why having to functions would be an improvement.